src/Microsoft.AspNet.Server.Kestrel/Http/Connection.cs

```diff
@@ -142,6 +156,22 @@ private void OnRead(UvStreamHandle handle, int status)
             _rawSocketInput.IncomingComplete(readCount, error);
         }
 
+        private Frame CreateFrame()
+        {
+            var frame = new Frame(this);
+            var httpConnectionFeature = frame as IHttpConnectionFeature;
+            if (httpConnectionFeature != null)
+            {
+                httpConnectionFeature.RemoteIpAddress = _remoteIPAddress;
+                httpConnectionFeature.RemotePort = _remotePort;
+                httpConnectionFeature.LocalIpAddress = _localIPAddress;
+                httpConnectionFeature.LocalPort = _localPort;
+                httpConnectionFeature.IsLocal = IPAddress.IsLoopback(_remoteIPAddress);
```

Comment:
We've done this check differently elsehwere: https://github.com/aspnet/WebListener/blob/cecd42dfcd624c322be43e4158163b5ac3ec4541/src/Microsoft.Net.Http.Server/RequestProcessing/Request.cs#L301